cli/src/commands/fix.rs

```diff
@@ -328,3 +382,111 @@ fn run_tool(
+    let mut tools_config = ToolsConfig { tools: Vec::new() };
+    if let Ok(tool_command) = config.get("fix.tool-command") {
+        // This doesn't change the displayed indices of the `fix.tools` definitions, and
+        // doesn't have a `name` that could conflict with them. That would matter more
+        // if we already had better error handling that made use of the `name`.
+        tools_config.tools.push(ToolConfig {
+            command: tool_command,
+            matcher: Box::new(EverythingMatcher),
+        });
+    }
+    if let Ok(tools_table) = config.get_table("fix.tools") {
+        // Convert the map into a sorted vector early so errors are deterministic.
+        let mut tools: Vec<ToolConfig> = tools_table
+            .into_iter()
+            .sorted_by(|a, b| a.0.cmp(&b.0))
+            .map(|(name, value)| -> Result<ToolConfig, CommandError> {
+                let tool: RawToolConfig = value.try_deserialize()?;
+                if tool.patterns.is_empty() {
+                    Err(config::ConfigError::Message(format!(
+                        "`fix.tools.{}.patterns` must not be empty. Use patterns=['all()'] if you \
+                         want to run this tool on every file.",
+                        name
+                    ))
+                    .into())
+                } else {
+                    Ok(ToolConfig {
+                        command: CommandNameAndArgs::Vec(
+                            NonEmptyCommandArgsVec::try_from(tool.command).map_err(|e| {
+                                config::ConfigError::Message(format!(
+                                    "`fix.tools.{}.command`: {}",
+                                    name, e
+                                ))
+                            })?,
+                        ),
+                        matcher: FilesetExpression::union_all(
+                            tool.patterns
+                                .iter()
+                                .map(|arg| {
+                                    fileset::parse(
+                                        arg,
+                                        // TODO: This is probably a misuse of Fs, but there is
+                                        // currently no other option for non-cwd-relative fileset
+                                        // parsing.
+                                        &RepoPathUiConverter::Fs {
+                                            cwd: workspace_command.workspace_root().clone(),
+                                            base: workspace_command.workspace_root().clone(),
+                                        },
+                                    )
+                                })
+                                .try_collect()?,
+                        )
+                        .to_matcher(),
+                    })
+                }
+            })
+            .try_collect()?;
+        tools_config.tools.append(&mut tools);
+    }
```

Comment:
Nit: Add a deprecation warning here.

Reply:
After we successfully parse the old syntax, there should be a warning like `"fix.tool-command is deprecated and will be removed in a future version, use fix.tool.<name> instead."`

Reply:
Oh, good idea. Thanks. Please take another look.

Reply:
This is great.